test/integration/sequelize.test.js

```diff
@@ -411,6 +411,115 @@ describe(Support.getTestDialectTeaser('Sequelize'), function() {
     }
   });
 
+  describe('set', function() {
+    it("should be configurable with global functions", function() {
+      var defaultClassMethod = sinon.spy()
+        , overrideClassMethod = sinon.spy()
+        , defaultInstanceMethod = sinon.spy()
+        , overrideInstanceMethod = sinon.spy()
+        , defaultSetterMethod = sinon.spy()
+        , overrideSetterMethod = sinon.spy()
+        , defaultGetterMethod = sinon.spy()
+        , overrideGetterMethod = sinon.spy()
+        , customClassMethod = sinon.spy()
+        , customOverrideClassMethod = sinon.spy()
+        , customInstanceMethod = sinon.spy()
+        , customOverrideInstanceMethod = sinon.spy()
+        , customSetterMethod = sinon.spy()
+        , customOverrideSetterMethod = sinon.spy()
+        , customGetterMethod = sinon.spy()
+        , customOverrideGetterMethod = sinon.spy();
+
+      this.sequelize.options.define = {
+        'classMethods': {
+          'defaultClassMethod': defaultClassMethod,
+          'overrideClassMethod': overrideClassMethod
+        },
+        'instanceMethods': {
+          'defaultInstanceMethod': defaultInstanceMethod,
+          'overrideInstanceMethod': overrideInstanceMethod
+        },
+        'setterMethods': {
+          'default': defaultSetterMethod,
+          'override': overrideSetterMethod
+        },
+        'getterMethods': {
+          'default': defaultGetterMethod,
+          'override': overrideGetterMethod
+        }
+      };
+      var testEntity = this.sequelize.define('TestEntity', {}, {
+        'classMethods': {
+          'customClassMethod': customClassMethod,
+          'overrideClassMethod': customOverrideClassMethod
+        },
+        'instanceMethods': {
+          'customInstanceMethod': customInstanceMethod,
+          'overrideInstanceMethod': customOverrideInstanceMethod
+        },
+        'setterMethods': {
+          'custom': customSetterMethod,
+          'override': customOverrideSetterMethod
+        },
+        'getterMethods': {
+          'custom': customGetterMethod,
+          'override': customOverrideGetterMethod
+        }
+      });
+
+      // Call all Class Methods
+      testEntity.defaultClassMethod();
+      testEntity.customClassMethod();
+      testEntity.overrideClassMethod();
+
+      expect(typeof testEntity.defaultClassMethod).to.equal('function');
+      expect(typeof testEntity.customClassMethod).to.equal('function');
+      expect(typeof testEntity.overrideClassMethod).to.equal('function');
+
+      expect(defaultClassMethod).to.have.been.calledOnce;
+      expect(customClassMethod).to.have.been.calledOnce;
+      expect(overrideClassMethod.callCount).to.be.eql(0);
```

Comment:
I tried. There was a strange error. It said something like "spy has to be a spy". (I'm on my mobile phone right now..)

Sent from my iPhone

> On 03.03.2015, at 19:18, Mick Hansen <notifications@github.com> wrote:
> 
> In test/integration/sequelize.test.js:
> 
> > +          'override': customOverrideGetterMethod
> > +        }
> > +      });
> > +
> > +      // Call all Class Methods
> > +      testEntity.defaultClassMethod();
> > +      testEntity.customClassMethod();
> > +      testEntity.overrideClassMethod();
> > +
> > +      expect(typeof testEntity.defaultClassMethod).to.equal('function');
> > +      expect(typeof testEntity.customClassMethod).to.equal('function');
> > +      expect(typeof testEntity.overrideClassMethod).to.equal('function');
> > +
> > +      expect(defaultClassMethod).to.have.been.calledOnce;
> > +      expect(customClassMethod).to.have.been.calledOnce;
> > +      expect(overrideClassMethod.callCount).to.be.eql(0);
> Can be express as .not.to.have.been.called i think.
> 
> —
> Reply to this email directly or view it on GitHub.
>